### PR TITLE
Manually place benchmark functions into the ITIM

### PR DESCRIPTION
**Pull request by nategraff-sifive**
Use -ffunction-section -marked sections to place benchmark code into the ITIM.

### CODE REVIEW

File: templates/base.lds [hunk 1]
```diff
@@ -211,23 +200,31 @@ SECTIONS
...
-         * of all post-init and non-constructor application code. Note that
-         * this may cause the program to fail to link if the ITIM is not large
-         * enough to contain the entire .text section. */
-        *(.text.unlikely .text.unlikely.*)
-        *(.text.startup .text.startup.*)
-        *(.text .text.*)
-        *(.gnu.linkonce.t.*)
-{% endif %}
         *(.itim .itim.*)
+{% block force_itim %}
```
Comment (bsousi5) [inline templates/base.lds:204]: I assume force_itim is a just a label for the block?

File: templates/ramrodata.lds [hunk 2]
```diff
@@ -15,3 +15,76 @@
...
  #}
 {% set ramrodata = True %}
+
+{% block force_itim %}
+{% if text_in_itim %}
+        /* The following takes advantage of -ffunction sections to link benchmark
+         * code into the ITIM when the ITIM is big enough to take advantage of it.
+         */
+
+        /* Dhrystone */
```
Comment (bsousi5) [inline templates/ramrodata.lds:25]: Creative maneuvering for the codes.

File: templates/ramrodata.lds [hunk 3]
```diff
@@ -15,3 +15,76 @@
  # Set layout options
  #}
 {% set ramrodata = True %}
+
+{% block force_itim %}
+{% if text_in_itim %}
+        /* The following takes advantage of -ffunction sections to link benchmark
```
[the remark this answers is not in the document]
Comment (nategraff-sifive) [reply templates/ramrodata.lds:21]: For now, we'll only be able to support these benchmarks using -ffunction-sections. My two-stage link idea will work for -fno-function-sections, but I'll need to implement that when I have more time to wrangle Coremark's Makefiles.

File: templates/base.lds (hunk 1 above)
Comment (nategraff-sifive) [reply templates/base.lds:204]: Yeah, the `force_itim` block gets substituted into the .itim section in base.lds.

File: templates/ramrodata.lds (hunk 3 above)
[the remark this answers is not in the document]
Comment (bsousi5) [reply templates/ramrodata.lds:21]: sound good.

**Pull request merged**